numba/targets/arrayobj.py

```diff
@@ -24,23 +24,8 @@ def make_array(array_type):
     Return the Structure representation of the given *array_type*
     (an instance of types.Array).
     """
-    dtype = array_type.dtype
```

Comment:
I wonder if "struct_factory" could now be done automatically by the datamodel system?